### ISSUE TITLE
Disconnecting from server and attempting to reconnect crashes Retroarch

### ISSUE THREAD

**Issue by dasbootbee** [libretro/flycast#629]
Retroarch 1.7.8.  Core just updated (IDK how to see the version)

Network configured using Quake or Planetweb disc
Connect to the network using a game (either game lobby or game site)
Disconnect (or be disconnected)
Try to reconnect -- Retroarch crashes

**Comment by flyinghead** [comment]
I can reproduce this issue with the regular flycast core but not with the flycast wince core.

The flycast wince core has many improvements and bug fixes and will eventually replace the regular core. You might want to check it out.

**Comment by dasbootbee** [comment]
Everything works as expected on the WinCE core.  Just a quick question: I followed the instructions on the Retroarch Quake online video (not sure where the network config stuff is documented besides there?) and I discovered some odd issues.  I would try to connect using something other than quake (Sonic Adventure or Chu Chu rocket being two examples) and it would try to connect and then crash the emulator.  Not until I used a Planetweb disc and configured the Dreamcast to use the dreampipe.net DNS would things work correctly.  Once I did that step then I could connect to the in-game web portal for everything that they re-created on Dreampipe.  Is that expected behavior?  I wasn't sure if you had some hard coded DNS hackery going on under the hood for the network stuff to work.  I was testing with the non-CE core so I don't know if this is a moot point now.

**Comment by flyinghead** [comment]
The crashes were probably due to bugs that have been fixed on the WinCE core. There is nothing to configure in terms of DNS: flycast is using the Dreamcast live DNS under the hood.

Actually, the ISP information is now automatically added to the flash ram by the emulator (if not entered yet). You can try it out by deleting (or just hiding) the dc_flash.bin and dc_nvmem.bin files. The former is no longer needed and the latter will be recreated at the next startup.

**Comment by dasbootbee** [comment]
Yes, that does indeed work.  I just nuked those two files and I had a connection profile in there to use when I fired up a game.  That brings up another issue.  Even with the new CE core, playing a peer to peer game like NFL2K2 or NBA2K2 game with another person (both users I tested with were on real Dreamcasts using DreamPi) we got fatal network errors and disconnects <5 minutes into the game each time.  Do you want a separate issue for that or is this a known problem?  Quake (peer to server) works fine from the limited amount of testing I did.

**Comment by flyinghead** [comment]
Many Dreamcast online games require some network ports to be open and forwarded to the emulator. This is the list for the 2K sport games series:
```
NBA/NFL/NCAA 2K Series: UDP 5502 | UDP 5503 | UDP 5656 | TCP 5011 | TCP 6666
```
One solution is to put the PC running the emulator in the DMZ zone of the router, which is what most DreamPi users do I suppose.

If you don't open these ports, you may be able to start a game but you will be disconnected after 1 min or so.

**Comment by dasbootbee** [comment]
I'll have to test with some more people then.  I had my PC in the DMZ and checked with the guy I was testing with and he said his DC was in the DMZ.

**Comment by flyinghead** [comment]
Assuming this is resolved. Reopen as needed
